cubed/array_api/creation_functions.py

```diff
@@ -25,6 +26,8 @@ def arange(
     num = int(max(math.ceil((stop - start) / step), 0))
     if dtype is None:
         dtype = nxp.arange(start, stop, step * num if num else step).dtype
+        dtype = nxp.dtype(f'{dtype.kind}{p}')
```

Comment:
This is a good example of where we should be using the inspection API rather than calling the function to find its dtype and then throwing away the result!